app/src/main/java/ch/epfl/cs311/wanderwave/di/ConnectionModule.kt

```diff
@@ -0,0 +1,35 @@
+package ch.epfl.cs311.wanderwave.di
+
+import android.content.Context
+import ch.epfl.cs311.wanderwave.model.remote.BeaconConnection
+import ch.epfl.cs311.wanderwave.model.remote.ProfileConnection
+import ch.epfl.cs311.wanderwave.model.remote.TrackConnection
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object ConnectionModule {
+
+  @Provides
+  @Singleton
+  fun provideBeaconConnection(@ApplicationContext context: Context): BeaconConnection {
+    return BeaconConnection()
+  }
+
+  @Provides
+  @Singleton
+  fun provideTrackConnection(@ApplicationContext context: Context): TrackConnection {
+    return TrackConnection()
+  }
+
+  @Provides
+  @Singleton
+  fun provideProfileConnection(@ApplicationContext context: Context): ProfileConnection {
+    return ProfileConnection()
+  }
+}
```

Comment:
It's a bit nitpicking about the naming, but I would still like to see a sort of repository structure instead of things directly using a "Connection". Then the connections could just implement the repository interface the way the local repositories did before. When things depend only on an abstract interface, it's much easier to add something like a caching layer without the naming becoming very confusing.
we could then have @Provides functions of type XYZRepository still returning the corresponding connection for now.